auto_submit/lib/requests/github_webhook.dart

```diff
@@ -25,19 +33,432 @@ class GithubWebhook {
     final String rawBody = await request.readAsString();
     final body = json.decode(rawBody) as Map<String, dynamic>;
 
-    if (!body.containsKey('pull_request') || !body['pull_request'].containsKey('labels')) {
+    if (!body.containsKey('pull_request') ||
+        !body['pull_request'].containsKey('labels')) {
       return Response.ok(jsonEncode(<String, String>{}));
     }
 
     PullRequest pullRequest = PullRequest.fromJson(body['pull_request']);
-    hasAutosubmit = pullRequest.labels!.any((label) => label.name == 'autosubmit');
+    hasAutosubmit =
+        pullRequest.labels!.any((label) => label.name == 'autosubmit');
 
     if (hasAutosubmit) {
-      // TODO(kristinbi): Check if PR can be submitted. https://github.com/flutter/flutter/issues/98707
+      // TODO(kristin): hardcode the githubToken this time for test, will add it to env later.
+      // final String githubToken = Platform.environment['AUTOSUBMIT_TOKEN']!;
+
+      final String githubToken = 'f4d8bc081a5f3ad57f3df3a99ec0417b269cec90';
+      final GithubService gitHub =
+          config.createGithubServiceWithToken(githubToken);
+
+      if (!(body.containsKey('repository') &&
+          body['repository'].containsKey('full_name') &&
+          body.containsKey('number'))) {
+        return Response.ok(jsonEncode(<String, String>{}));
+      }
+      RepositorySlug slug =
+          RepositorySlug.full(body['repository']['full_name']);
+      int number = body['number'];
```

Comment:
Are these variables expected to change later? Otherwise specify `final` keyword.

Reply:
Yes, thanks Keyong, I will add final to all variables which are not supposed change later.